scripts/verify-pr.py

```
@@ -0,0 +1,608 @@
+
+    if not pr_identifier or not pr_identifier.strip():
+        raise ValueError("PR identifier cannot be empty")
+
+    if not re.match(valid_pattern, pr_identifier):
+        raise ValueError(
+            f"Invalid PR identifier format: {pr_identifier!r}. "
+            "Allowed: PR numbers (7), branch names (feature/branch), "
+            "or comparisons (main...feature)"
+        )
+
+    logger.info(f"Validated PR identifier: {pr_identifier}")
```

Ruff (0.14.4)

Logging statement uses f-string

(G004)